### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by chris-crone**
* Python 3.7 to 3.9
* Linux: bump Debian base image from stretch to buster (required for Python 3.9)
* macOS: OpenSSL 1.1.1g to 1.1.1h, Python 3.7.7 to 3.9.0
* pyinstaller 3.6 to 4.1
* Loosen restriction on base images to latest minor

### CODE REVIEW

File: script/build/windows.ps1 [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #
 #        http://git-scm.com/download/win
 #
-# 2. Install Python 3.7.x:
+# 2. Install Python 3.9.x:
 #
 #        https://www.python.org/downloads/
 #
```
Comment (aiordache) [inline script/build/windows.ps1:12]: There are more py37 refs below in this file.

**Comment by ulyssessouza** [comment]
Needs to bump `pyinstaller` [here](https://github.com/docker/compose/blob/master/script/build/linux-entrypoint#L27) too

File: script/build/windows.ps1 (hunk 1 above)
Comment (chris-crone) [reply script/build/windows.ps1:12]: We'll need to make sure that we bump the CI machine's Python version as well

**Review by ulyssessouza — APPROVED** [review]
LGTM (with fear) :smile:

**Review by aiordache — APPROVED** [review]
LGTM :grin:

**Pull request merged**